pkg/testing/define/define_flags.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package define
+
+import (
+	"flag"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+type optionalBoolFlag struct {
+	set   bool
+	value bool
+}
+
+func (o *optionalBoolFlag) String() string {
+	if !o.set {
+		return "<not set>"
+	}
+	return strconv.FormatBool(o.value)
+}
+
+func (o *optionalBoolFlag) Set(s string) error {
+	o.set = true
+	if s == "" || s == "true" {
+		o.value = true
+		return nil
+	}
+	o.value = false
+	return nil
+}
+
+func (o *optionalBoolFlag) HasBeenSet() bool {
```

Comment:
I it _has been set_, then it _is set_, so why not make it simpler and use `IsSet` instead?